### PR TITLE
cdc, cmd: allow customized changefeed id

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Allow customized changefeed id.

### What is changed and how it works?

Create changefeed with an argurment `--changefeed` for `-c` to set its name.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has persistent data change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

* Support customize `changefeed` name.

### CODE REVIEW

File: cdc/model/changefeed.go [hunk 1]
```diff
@@ -78,6 +79,17 @@ type ChangeFeedInfo struct {
 	Error    *RunningError         `json:"error"`
 }
 
+var changeFeedIDRe *regexp.Regexp = regexp.MustCompile(`^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$`)
+
+// ValidateChangefeedID returns true if the changefeed ID matches
+// the pattern "^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$", eg, "simple-changefeed-task".
+func ValidateChangefeedID(changefeedID string) error {
+	if !changeFeedIDRe.MatchString(changefeedID) {
+		return errors.Errorf(`bad changefeed id, please match the pattern "^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$", eg, "simple-changefeed-task"`)
```
Comment (overvenus) [inline cdc/model/changefeed.go:88]:
```suggestion
		return errors.New(`bad changefeed id, please match the pattern "^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$", eg, "simple-changefeed-task"`)
```

File: cdc/kv/etcd.go [hunk 2]
```diff
@@ -197,6 +198,29 @@ func (c CDCEtcdClient) GetCaptures(ctx context.Context) (int64, []*model.Capture
...
+	value, err := info.Marshal()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	resp, err := c.Client.Txn(ctx).If(
+		clientv3.Compare(clientv3.ModRevision(key), "=", 0),
+	).Then(
+		clientv3.OpPut(key, value),
+	).Commit()
+	if !resp.Succeeded {
```
Comment (amyangfei) [inline cdc/kv/etcd.go:216]: Better to check `err` first. If `err` is not nil, `resp` is nil

File: cdc/model/errors.go [hunk 3]
```diff
@@ -20,15 +20,16 @@ import (
...
-	ErrTaskStatusNotExists   = errors.New("task status not exists")
-	ErrTaskPositionNotExists = errors.New("task position not exists")
-	ErrDecodeFailed          = errors.New("decode failed")
-	ErrAdminStopProcessor    = errors.New("stop processor by admin command")
-	ErrExecDDLFailed         = errors.New("exec DDL failed")
-	ErrCaptureNotExist       = errors.New("capture not exists")
-	ErrUnresolved            = errors.New("unresolved")
+	ErrWriteTsConflict         = errors.New("write ts conflict")
+	ErrChangeFeedNotExists     = errors.New("changefeed not exists")
+	ErrChangeFeedAlreadtExists = errors.New("changefeed already exists")
```
Comment (amyangfei) [inline cdc/model/errors.go:25]: s/ErrChangeFeedAlreadtExists/ErrChangeFeedAlreadyExists

**Comment by overvenus** [comment]
PTAL

**Comment by leoppro** [comment]
@amyangfei could you comment 'LGTM' again to trigger the merge robot, please?

**Pull request merged**